std/range/primitives.d

```diff
@@ -1986,6 +1992,146 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
     }
 }
 
+private enum int hasNothrowPopFront(Range) = (functionAttributes!(popFrontN!Range)
+                                               & FunctionAttribute.nothrow_);
+
+/**
+Pops the first element from a range or array $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+Params:
+    r = range to consume the first element from
+
+Warning:
+    If a transient range is used, you should `dup`licate it to avoid it from
+    being overwritten.
+    Using shift with `popFront` that might `throw` could lead to a loss of the
+    `front` element and thus it is prohibited.
+
+Returns:
+    Next element of the input range
+ */
+auto shift(Range)(ref Range r)
+if (isInputRange!Range && hasNothrowPopFront!Range)
```

Comment:
> You mean Andrei prefers the less common style. OK, can't argue with that.

Guys, I don't really care about this - the main point here is to get `shift` into Phobos :)
Btw we should just enable some code style tool like [DScanner](https://github.com/Hackerpilot/Dscanner) for Phobos and agree once on one of both ;-)